packages/terra-disclosure-manager/docs/README.md

```diff
@@ -0,0 +1,209 @@
+      >
+        <SlideGroup items={manager.disclosure.components} isAnimated />
+      </Modal>
+    </div>
+  )}
+>
+  <Child1 />
+  <Child2 />
+</DisclosureManager>
+
+```
+
+### Interacting with the Disclosure Manager
+
+#### Children
+
+The AppDelegate provided to the child components contains a `disclose` function. This `disclose` function validates the disclosure type with which it is provided against the set of supported disclosure types given to the DisclosureManager as a prop. If the provided type is not supported, and if the DisclosureManager was given an AppDelegate prop to fall back to, the DisclosureManager will call the disclose function provided by its AppDelegate prop.
+
+If the type is supported, the DisclosureManager will check the currently disclosured content's state to ensure it can be replaced. If the disclosure is denied, then `disclose` returns a rejected Promise. If the disclosure is allowed, then a resolved Promise is returned. This Promise will be resolved with an Object containing functions and Promises that can be used to manipulate the disclosure, if necessary. Included are `dismissDisclosure`, a function that will dismiss the disclosed content, as well as `afterDismiss`, a deferred Promise that will be resolved when the disclosed content is dismissed by any means. Alternatively, if the additional logic isn't needed, the returned Promise can be completely ignored.
+
+Example:
+```javascript
+app.disclose({
+  preferredType: 'disclosure-type',
+  size: 'large',
+  content: {
+    key: 'my-disclosed-content',
+    content: <DisclosedContent />,
+  }
+}).then({ dismissDisclosure, afterDismiss } => {
+  this.myContentIsDisclosed = true;
+
+  // This function can be cached and triggered as needed.
+  this.myDismissFunction = dismissDisclosure;
+
+  afterDismiss.then(() => {
+    // The afterDismiss will resolve when the content is dismissed, allowing for
+    // cleanup of local state as needed.
+    this.myContentIsDisclosed = false;
+
+    // If the dismissDisclosure function was cached, make sure it gets cleaned up.
+    this.myDismissFunction = false;
+  })
+}).catch((e) => {
+  // If this function is executed, the disclosure has been denied. You can use this to
+  // clean up any state that may be reliant on the disclosure ocurring.
+})
+```
+
+|Function|Description|
+|---|---|
+|`disclose(Object)`|Allows a component to disclose a component. This will open the disclosure if nothing was previously disclosed, or it will replace everything in the current disclosure stack. If disclosure was successful, it returns a Promise resolved with an Object containing the `dismissDisclosure` function and the `afterDismiss` deferred Promise. If disclosure was denied, it returns a rejected Promise.|
+
+`disclose` Argument API:
+
+|Key|Value|
+|---|---|
+|`preferredType`|The String describing the preferred disclosure type. This will be used to match the disclosure request to an appropriate DisclosureManager.|
```

Comment:
Lets add a sentence to the effect that the preferred type may not be the disclosure type used depending on availability and structure. ie if your component wants a slide panel disclosure but is embedded only in a modal manager, you get a modal.